Dockerfile

```diff
@@ -1,8 +1,20 @@
 FROM golang:1.24 AS builder
 WORKDIR /src
 COPY . .
-RUN make build
 
+# Detect platform and set architecture dynamically
+# Had to do this as I have a arm64 machine
+ARG TARGETPLATFORM
+RUN case "$TARGETPLATFORM" in \
+      "linux/amd64") export GOARCH=amd64 ;; \
+      "linux/arm64") export GOARCH=arm64 ;; \
+      *) echo "Unsupported platform: $TARGETPLATFORM" && exit 1 ;; \
+    esac && \
+    CGO_ENABLED=0 GOOS=linux go build -o /src/dicedb
```

Comment:
The [`build` target](https://github.com/DiceDB/dice/blob/master/Makefile#L7) contains a bunch of other params, can we add that here?

Also, now because of this change, the build logic is duplicated across two. Can we somehow keep it at one place? What are your thoughts?

Reply:
oh.. I think its better to use `make` instead here.. will update it.